added maps to preferences

jina2k opened the pull request:
# Description

Fixes #573 

## Type of change

- [ ] Note merging this changes the node modules
- [x] Note merging this changes the database configuration.
- [x] This change requires a documentation update

## Checklist

- [x] I have followed the [OED pull request](https://openenergydashboard.github.io/developer/pr.html) ideas
- [x] I have removed text in ( ) from the issue request

## Limitations
I have created a migrations folder with the relevant files to move to 0.7.1 as a step before 0.8.0. Documentation needs to be updated where there is an issue of opening localhost:3000 and seeing default map be "displayed" where it shows a blank image (occurs every time browser is reopened). Not that big of an issue as users can reload the page or click a random meter and it will correct itself.

jina2k committed:
added map to available default chart types

jina2k committed:
Merge branch 'development' of https://github.com/jina2k/OED into preferencemap

jina2k committed:
changed sql query to separate altering and create graph_type, now in migrations instead

jina2k committed:
migrations files

jina2k committed:
Merge branch 'development' into development

huss commented on src/server/migrations/0.7.0-0.7.1/sql/preferences/alter_preferences_type.sql:
```diff
@@ -0,0 +1,9 @@
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+DO $$ BEGIN
+    ALTER TYPE graph_type ADD VALUE 'map' AFTER 'compare';
```
I believe there is an issue that you cannot do this in a transaction which is what the migration code does. Thus, it must be done as in src/server/migrations/0.6.0-0.7.0/sql/meter/add_meter_type.sql.

jina2k replied:
changed lines to reflect add_meter_type.sql

huss replied:
Migrations are something we only recently got working correctly, esp. for changing an enum. Thus, I have pushed changes to get this to work as expected. I tested on my machine and the migration worked correctly. Note I did not push the version number change in package.json that is needed as this will be done when the release is made.

huss reviewed (changes requested):
The code works as desired and is good work. Just need to clean up the migration code (unless you show the current way works).

Also, would you like to open a new issue on the blank map or do you want someone else to do that? It just needs to be looked at sometime.

jina2k committed:
preferences

jina2k committed:
Merge branch 'development' of https://github.com/jina2k/OED into development

jina2k commented:
> The code works as desired and is good work. Just need to clean up the migration code (unless you show the current way works).
> 
> Also, would you like to open a new issue on the blank map or do you want someone else to do that? It just needs to be looked at sometime.

Someone else can open an issue for now, I will work on the group page in the meantime

huss committed:
fix up 0.7.1 migration

- This makes the alter of graph_type work with migration transactions.
- Moves index.js into proper directory.

huss commented:
> > The code works as desired and is good work. Just need to clean up the migration code (unless you show the current way works).
> > Also, would you like to open a new issue on the blank map or do you want someone else to do that? It just needs to be looked at sometime.
> 
> Someone else can open an issue for now, I will work on the group page in the meantime

I opened issue #667 to address this.
